x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/FoldablesConvertFunction.java

```diff
@@ -59,7 +59,8 @@ protected final TypeResolution resolveType() {
 
     @Override
     protected final Map<DataType, BuildFactory> factories() {
-        // TODO if a union type field is provided as an input, the correct error message is not shown, #112668 is a follow up
+        // This is used by ResolveUnionTypes, which is expected to be applied to ES fields only
+        // FoldablesConvertFunction takes only constants as inputs, so this is empty
```

Comment:
Similarly, this comment claims that this is used, but in `resolveConvertFunction` (line 1229), the check for `supportedTypes.isEmpty()` is always true, no?